modules/EvseV2G/connection/tls_connection.cpp

```diff
@@ -142,30 +141,34 @@ bool build_config(tls::Server::config_t& config, struct v2g_context* ctx) {
 
     // information from libevse-security
     const auto cert_info =
-        ctx->r_security->call_get_leaf_certificate_info(LeafCertificateType::V2G, EncodingFormat::PEM, false);
+        ctx->r_security->call_get_all_valid_certificates_info(LeafCertificateType::V2G, EncodingFormat::PEM, true);
     if (cert_info.status != GetCertificateInfoStatus::Accepted) {
         dlog(DLOG_LEVEL_ERROR, "Failed to read cert_info! Not Accepted");
     } else {
-        if (cert_info.info) {
-            const auto& info = cert_info.info.value();
-            const auto cert_path = info.certificate.value_or("");
-            const auto key_path = info.key;
-
-            // workaround (see above libevse-security comment)
-            const auto key_password = info.password.value_or("");
-
-            auto& ref = config.chains.emplace_back();
-            ref.certificate_chain_file = cert_path.c_str();
-            ref.private_key_file = key_path.c_str();
-            ref.private_key_password = key_password.c_str();
-
-            if (info.ocsp) {
-                for (const auto& ocsp : info.ocsp.value()) {
-                    const char* file{nullptr};
-                    if (ocsp.ocsp_path) {
-                        file = ocsp.ocsp_path.value().c_str();
+        if (!cert_info.info.empty()) {
+            // process all known certificate chains
+            for (const auto& chain : cert_info.info) {
+                const auto cert_path = chain.certificate.value_or("");
+                const auto key_path = chain.key;
+                const auto root_pem = chain.certificate_root.value_or("");
+
+                // workaround (see above libevse-security comment)
+                const auto key_password = chain.password.value_or("");
+
+                auto& ref = config.chains.emplace_back();
+                ref.certificate_chain_file = cert_path.c_str();
```

Comment:
actually i think this will eventually lead to a seg fault

the c_str is the same as data https://en.cppreference.com/w/cpp/string/basic_string/c_str
and data returns the pointer to raw memory of the stirng

if i read it correctly the cert_info and with it the cert_path are going out of scope once this function returns so the pointer you're accessing here will be invalid on function return

Reply:
certificate_chain_file is a ConfigItem which is a std::optional<std::string>.
`ref.certificate_chain_file = cert_path.c_str();` creates a new string object that is a copy of `cert_path.c_str()`.
ref.certificate_chain_file remains in scope after the function return as the config object is passed in by reference as a parameter.